server/src/main/java/com/flipkart/varadhi/db/ZKMetaStore.java

```diff
@@ -27,7 +29,8 @@ public <T> String create(T resource, int version, String resourcePath) {
             log.debug("Persisted Entity({}) in ZK: {}", resourcePath, response);
             return response;
         } catch (Exception e) {
-            throw new VaradhiException(e);
+            log.error(String.format("Failed to create entity %s.", resource), e);
+            throw new MetaStoreException(e);
```

Comment:
Add error message to MetaStoreException and remove log.error() -- this is the updated convention i.e. log only if being handled otherwise throw with required info.

This will be applicable to other methods as well.